### PR TITLE
Add authType to deploy, add function token to run

### PR DESCRIPTION
**Pull request by leerosen-cape**

### CODE REVIEW

**Comment by leerosen-cape** [comment]
@bendecoste CLI now up to date to compliment to merged runtime changes. (Verified they were backward compatible before merging)

File: cmd/cape/cmd/deploy.go [hunk 1]
```diff
@@ -91,9 +92,9 @@ func deploy(cmd *cobra.Command, args []string) error {
 		return UserError{Msg: "name not specified correctly", Err: err}
 	}
 
-	pcrSlice, err := cmd.Flags().GetStringSlice("pcr")
+	authType, err := cmd.Flags().GetString("authType")
```
Comment (bendecoste) [inline cmd/cape/cmd/deploy.go:95]: we should validate that you can't pass gibberish for the auth type

Comment (bendecoste) [reply cmd/cape/cmd/deploy.go:95]: runtime should also reject any unknown auth types if it doesn't already

**Review by bendecoste — CHANGES_REQUESTED** [review]

**Comment by leerosen-cape** [comment]
@bendecoste Validated this works and doesnt break anything. This requires the final supervisor entity rename to work in unison.

File: cmd/cape/cmd/deploy.go [hunk 2]
```diff
@@ -75,7 +76,7 @@ func init() {
 	rootCmd.AddCommand(deployCmd)
 
 	deployCmd.PersistentFlags().StringP("name", "n", "", "a name to give this function (default is the directory name)")
-	deployCmd.PersistentFlags().StringSliceP("pcr", "p", []string{""}, "pass multiple PCRs to validate against")
```
Comment (bendecoste) [inline cmd/cape/cmd/deploy.go:78]: why are you removing the PCR validation?

Comment (leerosen-cape) [reply cmd/cape/cmd/deploy.go:78]: @bendecoste Its ununsed, is this coming back? It was intelliJ cleaning up

File: cmd/cape/cmd/deploy.go [hunk 3]
```diff
@@ -43,6 +43,7 @@ type DeployResponse struct {
 }
 
 const storedFunctionMaxBytes = 1_000_000_000
+const defaultAuthType = "auth0"
```
Comment (bendecoste) [inline cmd/cape/cmd/deploy.go:46]: Should this be `entities.AuthenticationTypeAuth0` ?

File: cmd/cape/cmd/run.go [hunk 4]
```diff
@@ -142,23 +144,32 @@ func Run(url string, functionID string, file string, insecure bool) ([]byte, err
...
-	if err != nil {
-		return nil, err
+	if functionToken == "" {
+		userToken, err := getAuthToken()
+		if err != nil {
+			return nil, err
+		}
+		authToken = userToken
+	} else {
+		authProtocolType = "cape.function"
```
Comment (bendecoste) [inline cmd/cape/cmd/run.go:168]: the auth middleware in runtime doesn't let this be anything but cape.runtime. Is the plan to add support after this is merged?

**Review by bendecoste — DISMISSED** [review]

File: cmd/cape/cmd/run.go (hunk 4 above)
Comment (leerosen-cape) [reply cmd/cape/cmd/run.go:168]: We added runtime code to check for this and use the function token context. You already approved that PR.

**Review by bendecoste — APPROVED** [review]

**Pull request merged**